### ISSUE TITLE
Can't update any plugins

### ISSUE THREAD

**Issue by glasswork** [getgrav/grav-plugin-admin#2142]
I, all of a sudden, can't update any plugins: TypeError: Failed to fetch.

I was initially using Grav v1.6.30 but tried with all versions of grav through v1.7.14.

As per another ticket ( _Originally posted by @rhukster in https://github.com/getgrav/grav-plugin-admin/issues/1847#issuecomment-595800309_ ) I used every possible value (auto, fopen, cURL) for the Remote Fetch Method config.

This occurs in Chrome AND Firefox.

**Comment by glasswork** [comment]
I had managed to get the first bit, where is goes to dependencies to seem like it worked, but then it still failed, so opened this back up. I can update Grav, but Plugins OR Themes.

**Comment by mahagr** [comment]
Have you tried to update plugins from CLI?

**Comment by glasswork** [comment]
Very deliberately ... no ... since, while that might update the plugins, it would be unlikely to solve the issue.
I'd like to solve (and understand) the issue if I can ;)

But I will do it on one machine I have to test your hypothesis and see whether it is an Admin issue. 
BTW - on that score: I am running v1.9.1 of Admin.

**Comment by glasswork** [comment]
Yes, updating from CLi worked, so it IS likely to have been an admin issue.
And since it was , I suppose there is a chance that the update to v1.9.18 fixed the issue.
Sadly there are no further updates for the moment on which to test that theory, but I will try and manually bring over some old plugins later today for testing purposes.

**Comment by glasswork** [comment]
That being done, my site is broken in several ways which I will have to investigate.
One page I can't edit inside admin at all because it logs me out with every attempt to open it.
I can edit all the child pages of that page. 
I checked permissions and all seems fine on that front.
I am still getting the "Fetch Failed: The network connection was lost." when I try and clear the cache through the admin.
Tried all values available for the Remote Fetch Method.
So, plugins have been updated, but my Grav site is even less functional than it was ... not exactly the hoped for result ;)

**Comment by glasswork** [comment]
Hmmm ... for testing purposes I created a new Grav install with the latest version of Grav and the Admin plugin. And started (one-by-one) importing my plugins. All was going fine until I went into the admin plugin and set the Administrator Path to the value I have been using for years (e.g. not '/admin'), and the next old plugin I pulled over and tried to update failed in the way described above. When I changed the value back to '/admin' updating the plugins started to work again. Maybe there is a hardcoded path somewhere no longer taking that var for Administrator path into consideration?

**Comment by rhukster** [comment]
We develop on Mac mad have never run into this so i don’t think it’s a broad macOS issue.

**Comment by glasswork** [comment]
It is odd, for sure. I am going to play around some more and see if I can isolate the issue using the log from the two mac machines.

**Comment by mahagr** [comment]
Do you have any special characters in admin url?

**Comment by glasswork** [comment]
no, just lowercase letters

**Comment by glasswork** [comment]
1. Mojave machine **won't** work with Grav 1.7+ because it was running php7.3.26. Will fix that in a bit, but using it for testing.
2. Catalina machine **will** work with Grav 1.7+ because it has php7.3.6 , but only using default value for Administrator Path.

Still working through everything.

**Comment by glasswork** [comment]
Just incidentally, I can make a separate issue for this, but the admin needs a bit of css to contain its content:

![Screen Shot 2021-05-13 at 6 53 44 AM](https://user-images.githubusercontent.com/4681577/118131733-21f72280-b3bc-11eb-875e-7b3f75818874.png)

This is fixed by adding:

.form-list-wrapper ul[data-collection-holder] > li.collection-collapsed { overflow:hidden; }

Fix tested using the Customadmin plugin.

**Comment by mahagr** [comment]
@glasswork Please make a new issue about it. :)

**Comment by mahagr** [comment]
Grav should work perfectly with PHP 7.3. Which version of Grav are you on now?

**Comment by glasswork** [comment]
By the way: I have now updated an existing branch and also made a completely clean new download, and both have the same issues.

Grav works with php 7.3.6 (or so it says on the tin). It was running before I started upgrading on PHP 7.3.26 - although I would see the "failed to fetch" notices occasionally even then. Now, running v1.7.14 with all up-to-date plugins, it throws the "failed to fetch" error even when I ask it to look for updates, let alone do updates. I can still use cli to update. But I want to have full-functionality.

Also, for what it is worth, it no longer throws "failed to fetch" errors when doing any of the "clear cache" functions - which it used to always do before upgrading.

I am running it on a my own system at a .local domain (to get around Chrome's unwillingness to allow one to use .dev any longer). And it is **not** run locally as https.

I actually have grav work to do this morning so I will have to switch to another branch (without the updates) to get stuff done. I have not pulled all the updates into my master, nor will i until I figure out what is going on.

**Comment by mahagr** [comment]
After reading your messages again, I am wondering if your PHP crashes. Can you test updates again and the same time look if new entries appear to either Apache error logs or Grav logs?

**Comment by glasswork** [comment]
I do have a repeating php error in my Apache2 log (I've been meaning to check my error logs for some time but life and death happened).

[php7:error] [pid 8018] [client 127.0.0.1:58431] PHP Fatal error:  Grav\\Plugin\\Admin\\Admin::messages(): The script tried to execute a method or access a property of an incomplete object. Please ensure that the class definition "Grav\\Framework\\Session\\Messages" of the object you are trying to operate on was loaded _before_ unserialize() gets called or provide an autoloader to load the class definition in /Users/myuser/myfolder/mysite/user/plugins/admin/classes/admin.php on line 353

**Comment by glasswork** [comment]
That said, those are all old. Nothing I do causes any more errors. Last error was May 10th.

**Comment by glasswork** [comment]
If I add a new plugin, it says dependencies install successfuly but plugin will fail.

**Comment by glasswork** [comment]
On one machine I am getting the following in the pache2 log:
[Thu May 20 09:02:17.162586 2021] [core:notice] [pid 60537] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
[Thu May 20 09:02:43.663548 2021] [core:notice] [pid 60537] AH00052: child pid 60541 exit signal Segmentation fault (11)

**Comment by mahagr** [comment]
Try again but look if there are new entries in the log. We need to be sure that is the error that happens during the request.

Segmentation fault is always a PHP bug, the first one is really odd as the autoloader should find the file.

**Comment by glasswork** [comment]
Yep, that is it ... at least on the one machine.
Will try the other on Monday.
